homeassistant/components/landisgyr_heat_meter/const.py

```diff
@@ -0,0 +1,181 @@
+"""Constants for the Landis+Gyr Heat Meter integration."""
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import EntityCategory
+
+DOMAIN = "landisgyr_heat_meter"
+
+HEAT_METER_SENSOR_TYPES = (
+    SensorEntityDescription(
+        key="heat_usage_gj",
+        icon="mdi:fire",
+        name="Heat usage GJ",
+        native_unit_of_measurement="GJ",
```

Comment:
Home Asssistant doesn't know about GJ. Convert it to MWh and set this to `ENERGY_MEGA_WATT_HOUR` and the device_class to `SensorDeviceClass.ENERGY`.

Reply:
Mm interesting suggestion. That way it could also become part of the Energy dashboard, I guess? Although that shouldn't get mixed up with electricity usage. I've struggled with this, since GJ is really the energy unit for this device, also the way it's billed for example. Therefore I'd really like to keep the GJ, but perhaps I can introduce an extra attribute with the converted MWh value? (or perhaps kWh, since the usage is only about 1 or 2 GJ per month).

Reply:
I just noticed that gas usage supports MWh as well. This could be an option.
The heat meter is really a different device, not measuring electricity or gas. I'd really like for HA to support heat measuring units (that are common in the Netherlands for heating as an alternative for gas), but until it does, perhaps gas measured in MWh is supposed to be used for alternative sources like this? I'd really like having it in the Energy Dashboard, since I don't have gas in my house anyway. I'll add the MWh and test it myself for a while. I'll keep the GJ though, as explained.

Reply:
I've tested this and it seems to work, although it does generate some warnings in the energy dashboard. I still think this is an improvement though, since I can now see my heat usage over a period in the energy dashboard.
@emontnemery could you have another look if things look okay now? Thanks a lot!

Reply:
I don't think we need the same sensor twice, once in GJ and once in MWh. I suggest to only keep the MWh sensor, and just name it "Heat usage".

> it does generate some warnings in the energy dashboard

What warnings did you get?

Reply:
![SmartSelect_20220801-095016_Home Assistant.jpg](https://user-images.githubusercontent.com/48653141/182111210-1eae800d-dc74-4592-8f99-1d9ec2524f68.jpg)

It's in Dutch, but I guess you'll understand what it says.

Reply:
Yeah, that's pretty clear.
The problem should be fixed by https://github.com/home-assistant/core/pull/76016, maybe you can verify it?

Reply:
Great! I will check it.

Reply:
I've now (not committed yet) removed the _gj entity and removed the postfixes from the names. I've kept the postfixes in the key (which becomes the entity id), because I need a reference to the naming in the API. But perhaps there is another better way to store an external reference? I couldn't find one in the SensorEntityDescription. 

Applying https://github.com/home-assistant/core/pull/76016 removes the warning.

Reply:
Oh never mind, I was looking at cached data. The key is not used for the entity id's, so all is fine now. I'll run the tests and commit.

Reply:
Just committed this.